pkg/app/AppListingService.go

```diff
@@ -780,7 +784,16 @@ func (impl AppListingServiceImpl) setIpAccessProvidedData(ctx context.Context, a
 		}
 
 		if ciPipeline != nil && ciPipeline.CiTemplate != nil && len(*ciPipeline.CiTemplate.DockerRegistryId) > 0 {
-			dockerRegistryId := ciPipeline.CiTemplate.DockerRegistryId
+			artifact, err := impl.ciArtifactRepository.Get(appDetailContainer.CiArtifactId)
+			if err != nil {
+				impl.Logger.Errorw("error in fetching ci artifact", "ciArtifactId", appDetailContainer.CiArtifactId, "error", err)
+				return bean.AppDetailContainer{}, err
+			}
+			dockerRegistryId, err := impl.ciPipelineConfigReadService.GetDockerRegistryIdForCiPipeline(ciPipelineId, artifact)
+			if err != nil {
+				impl.Logger.Errorw("error in fetching docker registry id", "ciPipelineId", ciPipelineId, "error", err)
+				return bean.AppDetailContainer{}, err
+			}
 			appDetailContainer.DockerRegistryId = *dockerRegistryId
```

Comment:
this will panic as there are multiple places where we are sending nil, nil in GetDockerRegistryIdForCiPipeline. need to handle nil check either here or send apt response in the method. commenting on the problematic returns

Reply:
I could have sent nil, custom err instead (nil, nil) and handled that error accordingly. But I kept the function the same and handled the cases where DockerRegistryId could be nil since  I didn't want to change the behavior in CD trigger, which would have bigger testing scope